### DISCUSSION TITLE
Set field results when form inputs are created with values

### DISCUSSION THREAD

**Discussion by mattngu01** [charmbracelet/huh#586] [Q&A]
**Is your feature request related to a problem? Please describe.**
I am currently unit testing my TUI application using bubbletea. I found it unintuitive that when setting a field's value, you cannot immediately retrieve the results after creating a form.

```golang
func TestFilledForm(t *testing.T) {
	event := sampleEvent()
	f := filledEventForm(EventWrapper{&event})

	// huh forms do not set the results until processing nextFieldMsg through all fields
	for i := 0; i < 50; i++ {
		f.Update(f.NextField())
	}

	if f.GetString("summary") != event.Summary {
		t.Errorf("Summary: %v != %v", f.Get("summary"), event.Summary)
	}

	if f.GetString("description") != event.Description {
		t.Errorf("Description: %v != %v", f.Get("description"), event.Summary)
	}

	if f.GetString("start") != event.Start.DateTime {
		t.Errorf("Start: %v != %v", f.Get("start"), event.Start.DateTime)
	}

	if f.GetString("end") != event.End.DateTime {
		t.Errorf("End: %v != %v", f.Get("end"), event.End.DateTime)
	}

	if f.GetString("location") != event.Location {
		t.Errorf("Location: %v != %v", f.GetString("location"), event.Location)
	}
}
```

**Describe the solution you'd like**

Within `NewForm()`, I would like to set the results to any existing values for fields within it.

**Describe alternatives you've considered**

I can always call `NextField()` in the Form setup for my unit tests, but I figure this behavior might confuse some users in the future.

**Additional context**

I just want to make sure if there is a specific choice for forms to work this way. If it is not intended, I can take a stab at it. I'm learning Go right now, and I think it would be a good opportunity.

**Comment by bashbunni** [comment]
Hey @mattngu01 I'm not sure I'm fully understanding the issue here. You are looking to be able to pre-define form field values upon form creation, right? It looks like your test above is testing that Huh is storing the values to the right variables. I don't think that kind of test is needed for a third party project. If anything that behaviour should be tested in the Huh library itself.

That said, it's totally possible that I'm misunderstanding the problem here. I'd be happy to help however possible, so please let me know if this is helpful or if I'm missing something :)
